lib/ansiblelint/utils.py

```diff
@@ -237,13 +237,15 @@ def _taskshandlers_children(basedir, k, v, parent_type):
             append_children(th['import_playbook'], basedir, k, parent_type, results)
         elif 'import_tasks' in th:
             append_children(th['import_tasks'], basedir, k, parent_type, results)
-        elif 'import_role' in th:
+        elif 'include_role' in th or 'import_role' in th:
             th = normalize_task_v2(th)
-            results.extend(_roles_children(basedir, k, [th['action'].get('name')], parent_type,
-                                           main=th['action'].get('tasks_from', 'main')))
-        elif 'include_role' in th:
-            th = normalize_task_v2(th)
-            results.extend(_roles_children(basedir, k, [th['action'].get('name')],
+            # gets first property that is defined
+            role_name = next(
+                filter(bool, (th['action'].get(k, None) for k in ['name', 'role'])), None)
```

Comment:
Usually combining functional style with generator expressions is discouraged.

Reply:
TBH, I am was very close to make the use of 'role' key raise the same exception as a missing one because is not documented. This would made the filter not needed.